### DISCUSSION TITLE
binds improvements

### DISCUSSION THREAD

**Discussion by dantefromhell** [YaLTeR/niri#611] [Ideas]
As I'm progressing in my niri journey there's 2 ideas I would love to see for binds, especially to make the "Important Hotkeys" dialog more useable

1. Naming binds. 
Since I have a few binds that require to run them in a shell (i'm using `sh`), the hotkeys window now shows multiple entries named `sh`. It would be great to be able to name binds, so that the name is shown in the hotkeys list instead.

2. Configure the hotlist. Don't get me wrong it's super useful. Actually so useful that I want to be able to configure the hotlist over time. The values shown per default are super helpful to remember the basics of my setup, but as I'm getting better in my workflow the hotkeys I have most problems remember are the less frequently used ones.
A flag that allows me to control the content of the hotkeys for each bind would be super helpful. That way the dialog can "grow" along with me ;-)

Also huge thanks for making niri in the first place! It's soo much more joy to use than almost all other wayland based compositors

**Comment by YaLTeR** [comment]
Hey!

1. Yeah, I thought about it. I guess a property like name= or maybe description= on the bind would do.
2. I suppose something like show-in-hotkey-overlay=true/false?

> Also huge thanks for making niri in the first place! It's soo much more joy to use than almost all other wayland based compositors

Glad you're enjoying it :)

**Reply by dantefromhell** [reply]
Re 1. Yes, either sounds fine to me.

Re 2. Yes that's what I had in mind. Not sure how the overlay is build, but possibly the list of currently shown shortcut could default `show-in-hotkey-overlay=true` while other default to `false`. In case breaking existing behaviour is unwanted.

**Comment by YaLTeR** [comment]
Implemented in a605e7f6227448bbaa18932a251c9cbab9a07686

**Reply by dantefromhell** [reply]
Yeah, thanks a ton!
